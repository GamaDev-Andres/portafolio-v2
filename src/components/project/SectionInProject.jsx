import PropTypes from 'prop-types'
import { useState } from 'react'
import { ModalPortal } from '../modal/Modal'
const SectionInProject = ({ titulo, descripcion, arrImgs }) => {
  const [isOpenModal, setIsOpenModal] = useState(false)

  const closeModal = () => {
    setIsOpenModal(false)
  }
  const id = titulo.split(' ').join('')
  return (
    <section id={id} className="text-fondos scroll-m-14">
      <h3 className="uppercase text-blue-400 font-semibold text-2xl my-4">
        {titulo}
      </h3>
      <p>{descripcion}</p>
      <div className="flex flex-wrap pt-4 gap-4 items-center">
        {arrImgs.map((el) => (
          <img
            onClick={() => setIsOpenModal(el)}
            className="cursor-pointer hover:scale-[1.01] transition-transform duration-300 object-cover max-h-[400px]  mx-auto ring-1"
            src={el}
            alt="screenshot-proyecto"
            key={el}
          />
        ))}
      </div>
      {isOpenModal && (
        <ModalPortal isOpen={isOpenModal} closeModal={closeModal}>
          <img
            className="object-cover max-h-[80vh] max-w-full mx-auto ring-1"
            src={isOpenModal}
            alt="modal-screenshot-proyecto"
          />
        </ModalPortal>
      )}
    </section>
  )
}
SectionInProject.propTypes = {
  titulo: PropTypes.string.isRequired,
  descripcion: PropTypes.string.isRequired,
  arrImgs: PropTypes.array.isRequired
}
export default SectionInProject
