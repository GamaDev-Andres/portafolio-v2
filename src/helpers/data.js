export const proyectos = [
  {
    titulo: 'BARBERAPP',
    descripcion: `Aplicación realizada con el stack MERN y styled components,
     enfocada en la administración de una barbería en donde el cliente tiene 
     permitido agendar una cita con un barbero basándose en su perfil ,
      cortes o calificación de otros clientes.`,
    tecnologias: [
      'React',
      'Express',
      'StyledComponents',
      'MongoDb',
      'Reducers'
    ],
    url: 'https://barberappag.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/BarberApp-fe',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644874961/portafolio/barberapp/barberapp_sdpgwl.png',
    fotos: []
  },
  {
    titulo: 'NETBOOK',
    descripcion:
      'Mini clon de una red social realizado con React, Redux y firebase, me baso en ciertas funcionalidades de Facebook, tales como, publicaciones, likes, comentarios y perfiles individuales.',
    tecnologias: ['React', 'Firebase', 'Redux', 'Sass'],
    url: 'https://agnetbook.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/network-social-react',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644874998/portafolio/netbook/netbook_gus84s.png',
    fotos: []
  },
  {
    titulo: 'GALERIA',
    descripcion:
      'Simulación de una galería web realizada con React y Firebase , la aplicación se basa en la posibilidad de crear álbumes y tomarse fotos directamente en el álbum deseado y también tiene la posibilidad de buscar imágenes en una API y guardarlas en un álbum.',
    tecnologias: ['React', 'Firebase', 'StyledComponents'],
    url: 'https://practica-galeria.web.app/',
    codigo: 'https://github.com/GamaDev-Andres/react-galeria',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875042/portafolio/galeriareact/reactFirebase_xjiauw.png',
    fotos: []
  },
  {
    titulo: 'MIS PELIS',
    descripcion:
      'Proyecto de películas hecho con Javascript, Sass y webpack .En este proyecto me baso en un diseño de una pagina funcional, consulto la API IMDB que me provee información tanto de series como de películas con sus respectivos tráileres y sinopsis. Implemento buscadores y lista de favoritos.',
    tecnologias: ['Html', 'Sass', 'Webpack', 'Javascript'],
    url: 'https://afmispelisgi.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/PeliculasOMDB',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875064/portafolio/mispelis/imagepelisOMDB_cchblf.png',
    fotos: []
  },
  {
    titulo: 'FINDYOURGAMES',
    descripcion:
      'Proyecto realizado con Html, css, javascript y gulp. En este proyecto consulto una API de videojuegos y organizo su información en una interfaz que me permite hacer búsqueda y filtrado de los videojuegos disponibles.',
    tecnologias: ['Html', 'Sass', 'Gulp', 'Javascript'],
    url: 'https://findyourgames.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/FindYourGames',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875081/portafolio/findyourgames/proyecto1_vsncjy.png',
    fotos: []
  }
]

export const miDescripcion = {
  quienSoy: {
    parrafo1: `Mi nombre es Andres Gama, me apasiona encontrar grandes retos que me
          permitan crecer profesional y personalmente, la programación me
          permite enfocarme en esto.Actualmente tengo 21 años y sé que lograre
          que mis conocimientos sean participes de grandes, útiles y reconocidos
          proyectos.`,
    parrafo2: `En mí podras ver a un chico ansioso por aprender y dispuesto a
          afrontar complejas curvas de aprendizaje y gracias a mi constancia y
          disciplina veras a alguien creciendo constantemente.`
  },
  proceso: {
    parrafo1: `Inicie con pequeños cursos y blogs de programación en YouTube,
            posteriormente me di cuenta de que este mundo de verdad me
            interesaba y a pesar de que lo miraba complicado ... debía
            intentarlo entonces me aventuré a profundizar y empecé a conocer
            plataformas como Udemy, Platzi, MDN, etc...`,
    parrafo2: `He tenido diferentes dificultades en mi proceso, tanto para aprender
            como para aplicar lo aprendido, entre estas dificultades se destaca
            el hecho de enfrentarme a retos desconocidos como pueden ser
            tecnologias nuevas o profundizar más una tecnología para obtener el
            resultado deseado o uno más optimo, al final entendí que lo más
            lindo de esta profesión es que se garantiza un crecimiento
            constante.`
  },
  presentacion:
    'apasionado por el mundo del desarrollo y por crear aplicaciones e interfaces que aporten y brinden grandes experiencias.'
}

export const links = {
  contacto: [
    {
      href: 'https://drive.google.com/file/d/1IIUutqCGZL_KSN7NqE-JdRYQl_6Qaqhd/view',
      type: 'Resumé'
    }
  ]
}
