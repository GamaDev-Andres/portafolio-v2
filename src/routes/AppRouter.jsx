import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
