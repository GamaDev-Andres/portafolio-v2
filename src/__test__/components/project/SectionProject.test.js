/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import SectionInProject from '../../../components/project/SectionInProject'
import { proyectos } from '../../../helpers/data'
import AppRouter from '../../../routes/AppRouter'

describe('probando <SectionProject/>', () => {
  const proyecto = proyectos.find(project => project.detalles).detalles[0]
  test('debe montarse correctamente ', () => {
    const wrapper = Enzyme.mount(
      <SectionInProject titulo={proyecto.titulo} arrImgs={proyecto.fotos} descripcion={proyecto.data} />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h3').text().trim()).toBe(proyecto.titulo)
    expect(wrapper.find('p').text().trim()).toBe(proyecto.data)
  })
  test('no debe de mostrarse en la ruta "home"', () => {
    const wrapper = Enzyme.mount(
      <HelmetProvider>
        <MemoryRouter initialEntries={['/']}>
          <AppRouter />
        </MemoryRouter>
      </HelmetProvider>
    )
    expect(wrapper.find('SectionProject').exists()).toBe(false)
  })
})
