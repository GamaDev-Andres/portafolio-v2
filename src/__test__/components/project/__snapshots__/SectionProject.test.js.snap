// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <SectionProject/> debe montarse correctamente  1`] = `
<section
  className="text-fondos scroll-m-14"
  id="Barberoycliente"
>
  <h3
    className="uppercase text-blue-400 font-semibold text-2xl my-4"
  >
    Barbero y cliente
  </h3>
  <p>
    BARBERAPP nos da brinda la posibilidad de que tanto clientes como barberos tengan una cuenta. Un cliente podrá agendar citas con el barbero de su preferencia y por otro lado el barbero en su cuenta de BARBBERAPP podrá ver estas citas agendadas y decidir si aceptarlas o rechazarlas.
  </p>
  <div
    className="flex flex-wrap pt-4 gap-4 items-center"
  >
    <img
      alt="screenshot-proyecto"
      className="cursor-pointer hover:scale-[1.01] transition-transform duration-300 object-cover max-h-[400px]  mx-auto ring-1"
      key="https://res.cloudinary.com/dapa84kxy/image/upload/v1646015171/portafolio/barberapp/64f2c810-73b7-4dfd-9c12-07d35d36e862_txgsvq.jpg"
      onClick={[Function]}
      src="https://res.cloudinary.com/dapa84kxy/image/upload/v1646015171/portafolio/barberapp/64f2c810-73b7-4dfd-9c12-07d35d36e862_txgsvq.jpg"
    />
    <img
      alt="screenshot-proyecto"
      className="cursor-pointer hover:scale-[1.01] transition-transform duration-300 object-cover max-h-[400px]  mx-auto ring-1"
      key="https://res.cloudinary.com/dapa84kxy/image/upload/v1646153693/portafolio/barberapp/menu-user-desk_jtgfwl.png"
      onClick={[Function]}
      src="https://res.cloudinary.com/dapa84kxy/image/upload/v1646153693/portafolio/barberapp/menu-user-desk_jtgfwl.png"
    />
    <img
      alt="screenshot-proyecto"
      className="cursor-pointer hover:scale-[1.01] transition-transform duration-300 object-cover max-h-[400px]  mx-auto ring-1"
      key="https://res.cloudinary.com/dapa84kxy/image/upload/v1646015130/portafolio/barberapp/43985bb6-9674-48b4-9c1a-53ac5c329fbc_lstegh.jpg"
      onClick={[Function]}
      src="https://res.cloudinary.com/dapa84kxy/image/upload/v1646015130/portafolio/barberapp/43985bb6-9674-48b4-9c1a-53ac5c329fbc_lstegh.jpg"
    />
    <img
      alt="screenshot-proyecto"
      className="cursor-pointer hover:scale-[1.01] transition-transform duration-300 object-cover max-h-[400px]  mx-auto ring-1"
      key="https://res.cloudinary.com/dapa84kxy/image/upload/v1646153692/portafolio/barberapp/menu-barber-desk_p0d8gq.png"
      onClick={[Function]}
      src="https://res.cloudinary.com/dapa84kxy/image/upload/v1646153692/portafolio/barberapp/menu-barber-desk_p0d8gq.png"
    />
  </div>
</section>
`;
